docs/howto/write-and-structure-charm-code.md

```diff
@@ -269,6 +269,46 @@ def get_version() -> str:
     return response.text
 ```
 
+(handle-status)=
+## Handle status
+
+Your charm should report its status to Juju. The best approach depends on the charm's complexity:
+
+  - If units don't need to coordinate, use [`self.unit.status`](ops.Unit.status) to report the
+    status of each unit independently. For example:
+
+    ```python
+    def _on_start(self, event: ops.StartEvent):
+        """Handle start event."""
+        self.unit.status = ops.MaintenanceStatus("starting server")
+        # ...
+        # Start the workload and wait for it to be ready.
+        # ...
+        self.unit.status = ops.ActiveStatus()
+    ```
+
+    When your charm code sets `self.unit.status`, Ops immediately sends the unit status to Juju.
+
+    Juju aggregates the unit statuses into an overall application status.
+
+    ```{admonition} Best practice
+    :class: hint
+
+    In practice, it's often awkward to decide which status to report. Instead of writing
```

Comment:
I like the best practice notes to be as small as possible, so that they'll be easier to consume in a single list of items to check. What do you think about removing the first sentence?